### PR TITLE
Enable by default new scheduler

### LINKED ISSUE #12737: [New Scheduler] Provide infeasible tasks / pending tasks information to the dashboard.
### What is the problem?

cc @mfitton Do we have a dashboard test to check this? 

### Reproduction (REQUIRED)
Please provide a short code snippet (less than 50 lines if possible) that can be copy-pasted to reproduce the issue. The snippet should have **no external library dependencies** (i.e., use fake or mock data / environments):

If the code snippet cannot be run by itself, the issue will be closed with "needs-repro-script".

- [ ] I have verified my script runs in a clean environment and reproduces the issue.
- [ ] I have verified the issue also occurs with the [latest wheels](https://docs.ray.io/en/master/installation.html).

### PR DESCRIPTION
**Pull request by ericl**
## Why are these changes needed?

Enables the new scheduler feature flag and removes the old flagged travis builds.

Closes https://github.com/ray-project/ray/issues/12737

Parallel tests:
Closes https://github.com/ray-project/ray/pull/12998
Closes https://github.com/ray-project/ray/pull/12997

### CODE REVIEW

**Comment by ericl** [comment]
Seems to be passing all tests except setResource in java

File: src/ray/raylet/scheduling/cluster_task_manager.cc [hunk 1]
```diff
@@ -330,6 +333,39 @@ bool ClusterTaskManager::CancelTask(const TaskID &task_id) {
   return false;
 }
 
+void ClusterTaskManager::FillPendingActorInfo(rpc::GetNodeStatsReply *reply) const {
```
Comment (ericl) [inline src/ray/raylet/scheduling/cluster_task_manager.cc:336]: Fixes the dashboard test.

**Review by richardliaw — APPROVED** [review]
(rubber stamp)

**Comment by ericl** [comment]
Tests have passed 3x, merging.

**Pull request merged**